pkg/asset/machines/platform.go

```diff
@@ -0,0 +1,48 @@
+package machines
+
+import (
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/machines/aws"
+	awstypes "github.com/openshift/installer/pkg/types/aws"
+	"github.com/pkg/errors"
+)
+
+// platform holds information retrieved from the platform where it can
+// be shared between Master and Worker.
+type platform struct {
+	AWSZones []string
+}
+
+var _ asset.Asset = (*platform)(nil)
+
+// Dependencies returns all of the dependencies directly needed by the
+// platform asset.
+func (a *platform) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.ClusterID{},
+	}
+}
+
+// Generate queries for the cluster name from the user.
+func (a *platform) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	dependencies.Get(installConfig)
+
+	switch installConfig.Config.Platform.Name() {
+	case awstypes.Name:
+		var err error
+		a.AWSZones, err = aws.AvailabilityZones(installConfig.Config.Platform.AWS.Region)
+		if err != nil {
+			return errors.Wrap(err, "fetch availability zones")
+		}
+	default:
```

Comment:
Don't need an explicit default case that does nothing.

Reply:
> Don't need an explicit default case that does nothing.

[Obsolete now][1].

[1]: https://github.com/openshift/installer/pull/1212#issuecomment-469597418